### PR TITLE
add protobuf and some serilizing functions

### PR DESCRIPTION
**Pull request by zeqing-guo**

### CODE REVIEW

**Review by leventeliu — APPROVED** [review]
LGTM, approve.

File: sqlchain/tx.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package sqlchain
+
+import (
+	"github.com/thunderdb/ThunderDB/crypto/hash"
+	"github.com/thunderdb/ThunderDB/crypto/signature"
+)
+
+type TX struct {
```
Comment (covenantdb) [inline sqlchain/tx.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type TX should have comment or be unexported

File: sqlchain/tx.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	version int32
+	TxIn    []*TxIn
+	TxOut   []*TxOut
+}
+
+type TxIn struct {
```
Comment (covenantdb) [inline sqlchain/tx.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type TxIn should have comment or be unexported

File: sqlchain/tx.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	previousHash hash.Hash
+	Signee       *signature.PublicKey
+	Signature    *signature.Signature
+}
+
+type TxOut struct {
```
Comment (covenantdb) [inline sqlchain/tx.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type TxOut should have comment or be unexported

**Review by xq262144 — CHANGES_REQUESTED** [review]
commit comment typo

**Review by xq262144 — APPROVED** [review]

**Pull request merged**